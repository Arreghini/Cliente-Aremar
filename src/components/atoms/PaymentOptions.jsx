import React, { useState, useEffect } from 'react';
import DepositPayReservation from './DepositPayReservation';
import RemainingPayReservation from './RemainingPayReservation';
import TotalPayReservation from './TotalPayReservation';

const PaymentOptions = ({ reservation }) => {
  const [paymentStatus, setPaymentStatus] = useState(reservation.paymentStatus);

  useEffect(() => {
    if (reservation.paymentStatus) {
      setPaymentStatus(reservation.paymentStatus);
    }
  }, [reservation.paymentStatus]);

  return (
    <div className="space-y-4">
      {paymentStatus === 'pending' && (
        <>
          <DepositPayReservation reservation={reservation} />
          <TotalPayReservation reservation={reservation} />
        </>
      )}
      {paymentStatus === 'deposit_paid' && (
        <RemainingPayReservation reservation={reservation} />
      )}
      {paymentStatus === 'fully_paid' && (
        <p className="text-green-600 font-semibold">Reserva pagada completamente âœ…</p>
      )}
    </div>
  );
};

export default PaymentOptions;
