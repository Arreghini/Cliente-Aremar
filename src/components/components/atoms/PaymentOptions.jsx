import React from 'react';
import DepositPayReservation from './DepositPayReservation';
import TotalPayReservation from './TotalPayReservation';
import RemainingPayReservation from './RemainingPayReservation';

const PaymentOptions = ({ reservation }) => {
  const remainingAmount = reservation.totalPrice - (reservation.amountPaid || 0);

  return (
    <div>
      <h3 className="text-lg font-semibold mb-4">Opciones de Pago</h3>
      {reservation.status === "pending" && (
        <>
          <DepositPayReservation reservation={reservation} />
          <TotalPayReservation reservation={reservation} />
        </>
      )}
      {reservation.status === "confirmed" && remainingAmount > 0 && (
        <RemainingPayReservation reservation={reservation} />
      )}
    </div>
  );
};

export default PaymentOptions;