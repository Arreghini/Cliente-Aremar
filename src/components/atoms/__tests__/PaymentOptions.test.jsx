import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import { vi } from 'vitest';
import PaymentOptions from '../PaymentOptions';

// Mock de los subcomponentes
vi.mock('../DepositPayReservation', () => ({
  default: (props) => (
    <div data-testid="deposit-pay">DepositPayReservation {props.reservation.id}</div>
  ),
}));
vi.mock('../TotalPayReservation', () => ({
  default: (props) => (
    <div data-testid="total-pay">TotalPayReservation {props.reservation.id}</div>
  ),
}));
vi.mock('../RemainingPayReservation', () => ({
  default: (props) => (
    <div data-testid="remaining-pay">RemainingPayReservation {props.reservation.id}</div>
  ),
}));

describe('PaymentOptions component', () => {
  it('renders deposit and total options when status is pending', () => {
    const reservation = {
      id: 'res1',
      totalPrice: 500,
      amountPaid: 0,
      status: 'pending',
    };

    render(<PaymentOptions reservation={reservation} />);

    expect(screen.getByText(/Opciones de Pago/i)).toBeInTheDocument();
    expect(screen.getByTestId('deposit-pay')).toHaveTextContent('DepositPayReservation res1');
    expect(screen.getByTestId('total-pay')).toHaveTextContent('TotalPayReservation res1');
    expect(screen.queryByTestId('remaining-pay')).not.toBeInTheDocument();
  });

  it('renders remaining option when status is confirmed and there is balance left', () => {
    const reservation = {
      id: 'res2',
      totalPrice: 500,
      amountPaid: 200,
      status: 'confirmed',
    };

    render(<PaymentOptions reservation={reservation} />);

    expect(screen.getByText(/Opciones de Pago/i)).toBeInTheDocument();
    expect(screen.getByTestId('remaining-pay')).toHaveTextContent('RemainingPayReservation res2');
    expect(screen.queryByTestId('deposit-pay')).not.toBeInTheDocument();
    expect(screen.queryByTestId('total-pay')).not.toBeInTheDocument();
  });

  it('renders nothing extra when confirmed and fully paid', () => {
    const reservation = {
      id: 'res3',
      totalPrice: 500,
      amountPaid: 500,
      status: 'confirmed',
    };

    render(<PaymentOptions reservation={reservation} />);

    expect(screen.getByText(/Opciones de Pago/i)).toBeInTheDocument();
    expect(screen.queryByTestId('remaining-pay')).not.toBeInTheDocument();
    expect(screen.queryByTestId('deposit-pay')).not.toBeInTheDocument();
    expect(screen.queryByTestId('total-pay')).not.toBeInTheDocument();
  });

  it('always renders the title "Opciones de Pago"', () => {
    const reservation = {
      id: 'res4',
      totalPrice: 300,
      amountPaid: 0,
      status: 'pending',
    };

    render(<PaymentOptions reservation={reservation} />);
    expect(screen.getByRole('heading', { name: /Opciones de Pago/i })).toBeInTheDocument();
  });

  it('passes the correct reservation object to DepositPayReservation', () => {
    const reservation = {
      id: 'res5',
      totalPrice: 700,
      amountPaid: 0,
      status: 'pending',
    };

    render(<PaymentOptions reservation={reservation} />);
    expect(screen.getByTestId('deposit-pay')).toHaveTextContent('DepositPayReservation res5');
  });

  it('matches snapshot for a confirmed reservation with remaining balance', () => {
    const reservation = {
      id: 'res6',
      totalPrice: 1000,
      amountPaid: 400,
      status: 'confirmed',
    };

    const { container } = render(<PaymentOptions reservation={reservation} />);
    expect(container).toMatchSnapshot();
  });
});
