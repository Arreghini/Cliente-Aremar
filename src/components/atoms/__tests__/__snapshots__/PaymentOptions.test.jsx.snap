// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PaymentOptions component > matches snapshot for a confirmed reservation with remaining balance 1`] = `
<div>
  <div>
    <h3
      class="text-lg font-semibold mb-4"
    >
      Opciones de Pago
    </h3>
    <div
      data-testid="remaining-pay"
    >
      RemainingPayReservation 
      res6
    </div>
  </div>
</div>
`;
